x/ecocredit/core/genesis_test.go

```diff
@@ -0,0 +1,414 @@
+package core_test
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/regen-network/regen-ledger/x/ecocredit/core"
+)
+
+func TestValidateGenesis(t *testing.T) {
+	x := `{
+		"regen.ecocredit.v1.BatchBalance":[
+			{
+				"address":"gydQIvR2RUi0N1RJnmgOLVSkcd4=",
+				"batch_id":"1",
+				"tradable":"90.003",
+				"retired":"9.997"
```

Comment:
we could add escrowed balance here as a test case

Reply:
Looks like this might have been overlooked. We should include a test case with a positive escrowed amount.

Reply:
It also looks like you are missing a `BatchBalance` validation.

Reply:
Also `CreditType` and `BatchSupply` validation.

Reply:
> Also `CreditType` and `BatchSupply` validation.
 
Added validation for `CreditType`. Batch balance & supply validation is already covered by `ValidateGenesis`.